src/test/java/com/google/cloud/compute/v1/it/ITInstanceTest.java

```diff
@@ -42,42 +76,191 @@ public static void setUp() throws IOException {
-            .build();
-    Instance instance =
-        Instance.newBuilder()
-            .setName("mytestinstancetemplate")
-            .setMachineType(machineType)
-            .addDisks(disk)
+  public void getInstanceTest() {
+    assertInstanceDetails(getInstance());
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void listInstancesTest() {
+    ListInstancesHttpRequest instancesHttpRequest =
+        ListInstancesHttpRequest.newBuilder().setZone(PROJECT_ZONE_NAME.toString()).build();
+    for (Instance instance : instanceClient.listInstances(instancesHttpRequest).iterateAll()) {
+      if (instance.getName().equals(INSTANCE)) {
+        assertInstanceDetails(instance);
+      }
+    }
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void updateInstanceTest() {
+    List<String> requiredProperties =
+        Arrays.asList("name", "machineType", "disks", "networkInterfaces");
+    List<String> updatableProperties = Arrays.asList("metadata", "fingerprint", "description");
+    List<String> fieldMask =
+        Arrays.asList(
+            requiredProperties.get(0),
+            requiredProperties.get(1),
+            requiredProperties.get(2),
+            requiredProperties.get(3),
+            updatableProperties.get(0),
+            updatableProperties.get(1),
+            updatableProperties.get(2));
+
+    Instance instance = getInstance();
+    assertInstanceDetails(instance);
+
+    UpdateInstanceHttpRequest request =
+        UpdateInstanceHttpRequest.newBuilder()
+            .setMostDisruptiveAllowedAction("REFRESH")
+            .setMinimalAction(null)
+            .setInstance(FORMATTED_INSTANCE)
+            .setInstanceResource(
+                instance
+                    .toBuilder()
+                    .setDescription("updated description")
+                    .setFingerprint(instance.getFingerprint())
+                    .build())
+            .addAllFieldMask(fieldMask)
             .build();
+    if (instance.getStatus().equals("RUNNING")) {
+      instanceClient.updateInstance(request);
+      Instance updatedInstance = getInstance();
+      assertThat(updatedInstance.getDescription()).isNotEqualTo(instance.getDescription());
+      assertThat(updatedInstance.getDescription()).isEqualTo("updated description");
+    }
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void instanceStartStopTest() {
+    StopInstanceHttpRequest stopInstanceHttpRequest =
+        StopInstanceHttpRequest.newBuilder().setInstance(FORMATTED_INSTANCE).build();
+    Operation stopOperation = instanceClient.stopInstance(stopInstanceHttpRequest);
+    assertThat(stopOperation.getOperationType()).isEqualTo("stop");
+
+    StartInstanceHttpRequest startInstanceHttpRequest =
+        StartInstanceHttpRequest.newBuilder().setInstance(FORMATTED_INSTANCE).build();
+    Operation startOperation = instanceClient.startInstance(startInstanceHttpRequest);
+    assertThat(startOperation.getOperationType()).isEqualTo("start");
+  }
```

Comment:
It seems to me, that your test does not verify that the instance is really stopped/started
`assertThat(stopOperation.getOperationType()).isEqualTo("stop");` is not enough.

Reply:
done